packages/block-serialization-default-parser/parser.php

```diff
@@ -361,20 +361,38 @@ function next_token() {
 
 		list( $match, $started_at ) = $matches[ 0 ];
 
-		$length    = strlen( $match );
+		$length = strlen( $match );
+
+		// We know where the attrs start, now search until the close of the block comment.
+		$attrs_start      = $started_at + $length;
+		$closing_position = strpos( $this->document, '-->', $attrs_start );
```

Comment:
I'd think as far as any HTML parser goes, it would consider the comment as complete once it encounters `<!-- wp:foo {"id":824,"foo":"look at -->` ?

Reply:
`-->` is fundamentally unalloyed inside a comment so I wouldn't block any changes on the presumption that it could be there.

number 1: it breaks block structure
number 2: it breaks the HTML and the page will break entirely

Reply:
I understand both your points, but I wonder how easily this could break (e.g. if some plugin is allowing block attributes to be set via the server, whether via block API or (imagine a bulk-migration tool that fails to escape values properly) via database interactions). Shouldn't the parser be robust enough?

Reply:
in this case I don't think it's a "robust-enough" problem. if some block API or server-side plugin adds `-->` inside of an HTML comment it will be the entire post that's broken.

I'm trying to understand a situation where the parser could resole this without massively breaking other things.

we could include a JSON parser inside of the post parser; this would slow down parsing I'm pretty sure, but it wouldn't solve the problem that now we're allowing extremely bad and broken HTML in a post. I don't want to give any impression that it's okay to put `-->` inside an HTML comment and I'd rather let developers find out immediately rather than burying it deeper.

Reply:
No, you're right, I'm convinced we shouldn't be absorbing this concern.